cmd/nameset-backfill/main.go

```diff
@@ -0,0 +1,179 @@
+			return err
+		}
+		added += len(results)
+	}
+	return nil
+}
+
+func (b *backfiller) findCerts() ([]resultHolder, error) {
+	var allResults []resultHolder
+	for {
+		var results []resultHolder
+		_, err := b.dbMap.Select(
+			&results,
+			`SELECT c.serial, c.issued, c.expires, c.der FROM certificates AS c
+       LEFT JOIN fqdnSets AS ns ON c.serial=ns.serial
+       WHERE ns.serial IS NULL
+       ORDER BY c.issued DESC
+       LIMIT ?
+       OFFSET ?`,
+			1000,
+			len(allResults),
+		)
+		if err != nil {
+			return nil, err
+		}
+		if len(results) == 0 {
+			break
+		}
+		b.stats.Inc("db-backfill.fqdnSets.missing-found", int64(len(results)), 1.0)
+		allResults = append(allResults, results...)
+	}
+	return allResults, nil
+}
+
+func hashNames(names []string) []byte {
+	names = core.UniqueLowerNames(names)
+	hash := sha256.Sum256([]byte(strings.Join(names, ",")))
+	return hash[:]
+}
+
+func (b *backfiller) processResults(results []resultHolder) error {
+	numResults := len(results)
+	added := 0
+	for _, r := range results {
+		c, err := x509.ParseCertificate(r.DER)
+		if err != nil {
+			b.log.Err(fmt.Sprintf("Failed to parse certificate [serial: %s] retrieved from database: %s", r.Serial, err))
+			continue
+		}
+		err = b.dbMap.Insert(&core.FQDNSet{
+			SetHash: hashNames(c.DNSNames),
+			Serial:  r.Serial,
+			Issued:  r.Issued,
+			Expires: r.Expires,
+		})
+		if err != nil {
+			b.log.Err(fmt.Sprintf("Failed to add name set for %s to database: %s", r.Serial, err))
+			continue
+		}
+		added++
+		b.stats.Inc("db-backfill.fqdnSets.added", 1, 1.0)
+	}
+	if added < numResults {
+		return fmt.Errorf("Didn't add all name sets, %d out of %d failed", numResults-added, numResults)
+	}
+	return nil
+}
+
+func main() {
+	amqpURI := flag.String("amqpURI", "", "AMQP connection URI")
+	amqpURIFile := flag.String("amqpURIFile", "", "File to read AMQP connection URI from")
+	amqpCert := flag.String("amqpCert", "", "AMQP client certificate to use")
+	amqpKey := flag.String("amqpKey", "", "Key for AMQP client certificate")
+	amqpCA := flag.String("amqpCA", "", "Root CA to trust for AMQP connections")
+
+	statsdURI := flag.String("statsdURI", "", "StatsD URI")
+
+	dbConnect := flag.String("dbConnect", "", "DB connection URI")
+	dbConnectFile := flag.String("dbConnectFile", "", "File to read DB connection URI from")
+
+	syslogNet := flag.String("syslogNetwork", "", "Syslog network")
+	syslogURI := flag.String("syslogServer", "", "Syslog URI")
+	syslogLevel := flag.Int("syslogLevel", 7, "Level at which to log")
+	flag.Parse()
+
+	dbConf := cmd.DBConfig{DBConnect: *dbConnect, DBConnectFile: *dbConnectFile}
+	dbURI, err := dbConf.URL()
+
+	amqpConf := &cmd.AMQPConfig{
+		Server:        *amqpURI,
+		ServerURLFile: *amqpURIFile,
+		SA: &cmd.RPCServerConfig{
+			Server:     "SA.server",
+			RPCTimeout: cmd.ConfigDuration{Duration: time.Second * 15},
+		},
+	}
+	if *amqpCert != "" && *amqpKey != "" && *amqpCA != "" {
+		amqpConf.Insecure = false
```

Comment:
Insecure should already be set to false, right?
How about `else { amqpconf.Insecure = true }`